app/models/project.rb

```diff
@@ -449,53 +461,75 @@ def normalize_licenses
   end
 
   def update_repository_async
-    RepositoryProjectWorker.perform_async(self.id) if known_repository_host_name.present?
+    RepositoryProjectWorker.perform_async(id) if known_repository_host_name.present?
   end
 
   def known_repository_host_name
     github_name_with_owner || bitbucket_name_with_owner || gitlab_name_with_owner
   end
 
   def known_repository_host
-    return 'GitHub' if github_name_with_owner.present?
-    return 'Bitbucket' if bitbucket_name_with_owner
-    return 'GitLab' if gitlab_name_with_owner
+    return "GitHub" if github_name_with_owner.present?
+    return "Bitbucket" if bitbucket_name_with_owner
+    return "GitLab" if gitlab_name_with_owner
   end
 
   def can_have_dependencies?
     return false if platform_class == Project
+
     platform_class::HAS_DEPENDENCIES
   end
 
   def can_have_entire_package_deprecated?
     return false if platform_class == Project
+
     platform_class::ENTIRE_PACKAGE_CAN_BE_DEPRECATED
   end
 
   def can_have_versions?
     return false if platform_class == Project
+
     platform_class::HAS_VERSIONS
   end
 
   def release_or_tag
-    can_have_versions? ? 'releases' : 'tags'
+    can_have_versions? ? "releases" : "tags"
   end
 
   def update_repository
     return false unless known_repository_host_name.present?
+
     r = Repository.create_from_host(known_repository_host, known_repository_host_name)
     return if r.nil?
-    unless self.new_record?
+
+    unless new_record?
       self.repository_id = r.id
-      self.forced_save
+      forced_save
+    end
+  end
+
+  def update_tags
+    return unless repository
+
+    benchmark = nil
+
+    qcount = QueryCounter.count do
+      benchmark = Benchmark.measure do
```

Comment:
Could also use this if you want to avoid the multiplication below:

```suggestion
      ms = Benchmark.ms do
```